### PR TITLE
Recordset Errors

### PR DESCRIPTION
**Pull request by jrchudy**
Opening this pull request with my comments for `recordset`.

### CODE REVIEW

File: recordset/recordset.js [hunk 1]
```diff
@@ -79,10 +79,12 @@
         $scope.navbarBrandImage = (chaiseConfig['navbarBrandImage'] !== undefined? chaiseConfig.navbarBrandImage : "");
         $scope.navbarBrandText = (chaiseConfig['navbarBrandText'] !== undefined? chaiseConfig.navbarBrandText : "Chaise");
 
+        // TODO: Session.login should already be wrapped in a try catch, maybe don't need to wrap this function with the wrapper (Utils.catchWrapper(function() {...}))
         $scope.login = function() {
```
Comment (RFSH) [inline recordset/recordset.js]: I don't think `login` and `logout` functions will even be needed here. Isn't the navbar responsible for these functionalities?

File: recordset/recordset.js [hunk 2]
```diff
@@ -95,11 +97,13 @@
             $rootScope.location = $window.location.href;
         });
 
+        // TODO: We might not need to wrap functions if $on("error", ...) will catch errors thrown from one time function events
         $scope.$on('error', function(event, exception) {
```
Comment (RFSH) [inline recordset/recordset.js]: Does other apps need this event handler? It's only available in this app.

File: recordset/recordset.js (hunk 1 above)
Comment (jrchudy) [inline recordset/recordset.js]: I think you are correct.

File: recordset/recordset.js (hunk 2 above)
Comment (jrchudy) [inline recordset/recordset.js]: I noticed that in `ellipses.js` we are `emitting` errors when they occur and then calling the catchAll function. It's becoming more apparent to me that this is most likely unnecessary.

File: recordset/recordset.js [hunk 3]
```diff
@@ -219,6 +224,7 @@
                 $log.warn(exception);
```
Comment (RFSH) [inline recordset/recordset.js]: I think the whole catch should be replaced by `catchAll` (assuming that `catchAll` will eventually pop-up the error if it cannot handle it)

**Comment by chiragsanghvi** [comment]
@jrchudy @howdyjessie I have made the changes for global error handling as well as recordset app.

https://dev.gpcrconsortium.org/~chirag/chaise/recordset/#1/experiments:experiment

File: common/errorDialog.controller.js [hunk 4]
```diff
@@ -6,10 +7,21 @@
     .controller('ErrorDialogController', ['$uibModalInstance', 'params', function ErrorDialogController($uibModalInstance, params) {
         var vm = this;
         vm.params = params;
+        // TODO: Wrap ok function in catcher fn
```
Comment (RFSH) [inline common/errorDialog.controller.js]: @chiragsanghvi Is this comment still relevant?

File: common/errors.js [hunk 5]
```diff
@@ -44,6 +45,7 @@
                     $window.location.replace(redirectLink);
                 }
             });
+            // TODO: Add .catch for catchAll
```
Comment (RFSH) [inline common/errors.js]: @chiragsanghvi Same comment about here.

Comment (chiragsanghvi) [inline common/errors.js]: Removed the comment

File: common/errorDialog.controller.js (hunk 4 above)
Comment (chiragsanghvi) [inline common/errorDialog.controller.js]: Removed the comment

**Comment by RFSH** [comment]
Besides the two comments that I just made, I think the code looks good to merge. 

This is how the output of new catchAll looks like. Mentioning @hongsudt for comments about UI.
screen shot 2017-03-28 at 6 10 39 pm

**Comment by hongsudt** [comment]
@chiragsanghvi We should be displaying 
Error: Terminal Error 
An unexpected error has occurred. Please report this problem to your system administrators.
Click OK to return to the Home Page, or Show Details to see technical details associated with this error. 

There should be a terminal error object that contains this information, correct?

**Comment by chiragsanghvi** [comment]
@hongsudt This is how it will look for terminal errors. Other HTTP Error handling we will add later.

screen shot 2017-03-28 at 11 11 42 pm

**Comment by jrchudy** [comment]
The changes look good to me. The error is detailed enough and provides the information that we had talked about for stack traces.

**Pull request merged**